src/rules.jl

```diff
@@ -518,6 +519,43 @@ function lrp!(Rᵏ, rule::GeneralizedGammaRule, layer, modified_layers, aᵏ, R
     @. Rᵏ = aᵏ⁺ * (cˡ⁺ + cʳ⁻) + aᵏ⁻ * (cˡ⁻ + cʳ⁺)
 end
 
+"""
+#
+"""
+struct LayerNormRule <: AbstractLRPRule end
+is_compatible(::LayerNormRule, ::LayerNorm) = true
+
+function lrp!(
+    Rᵏ, ::LayerNormRule, layer::LayerNorm{F,D}, _modified_layer, aᵏ, Rᵏ⁺¹
+) where {F,D<:typeof(identity)}
+    n_dims = 1:length(layer.size)
+    μₐ = mean(aᵏ; dims=n_dims)
+    s = @. Rᵏ⁺¹ / stabilize_denom(aᵏ - μₐ, LRP_DEFAULT_STABILIZER)
+    μₛ = mean(s; dims=n_dims)
+    @. Rᵏ = aᵏ * (s - μₛ)
+end
+
+function lrp!(Rᵏ, ::LayerNormRule, layer::LayerNorm, _modified_layer, aᵏ, Rᵏ⁺¹)
+    layer_norm, scale = split_layer(layer)
+    eps = convert(float(eltype(aᵏ)), layer_norm.ϵ)
+    n_dims = 1:length(layer_norm.size)
+    μₐ = mean(aᵏ; dims=n_dims)
+    # forward pass: compute normalized inputs aᵏ ->(normalize) aᵏₙ 
+    z = aᵏ .- μₐ
+    σ = std(aᵏ; dims=n_dims, mean=μₐ, corrected=false)
+    aᵏₙ = @. z / (σ + eps)
+    # call ZeroRule on affine part as a fallback when model is not canonized
+    lrp!(Rᵏ, ZeroRule(), scale, modify_layer(ZeroRule(), scale), aᵏₙ, Rᵏ⁺¹)
+    # compute LRP pass through normalization
+    s = @. Rᵏ / stabilize_denom(z, LRP_DEFAULT_STABILIZER)
+    μₛ = mean(s; dims=n_dims)
+    @. Rᵏ = aᵏ * (s - μₛ)
+end
+
+function lrp!(Rᵏ, ::ZeroRule, layer::LayerNorm, modified_layer, aᵏ, Rᵏ⁺¹)
+    lrp!(Rᵏ, LayerNormRule(), layer, modified_layer, aᵏ, Rᵏ⁺¹)
+end
```

Comment:
I think we should remove this as it silently switches the `ZeroRule` for the `LayerRule`:
* either we exclusively allow the `LayerNorm` rule on `LayerNorm` layers 
* or we also allow the `ZeroRule` AD fallback on L12

Reply:
I don't have strong opinions on this, but I agree this needs to change, because my implementation does not allow to use the `ZeroRule` on `LayerNorm` at all, which is definitely a problem. Some things to consider:
1. Exclusively allow the `LayerNormRule`: this would make the default analyzer generated by `LRP` throw an error, because it assigns `ZeroRule` everywhere. So I guess we would need to change the default rule.
2. Allow the default fallback for `ZeroRule`: would be an option, but I think the "average" user maybe does not think too much about which rule to use for `LayerNorm`, and maybe giving them the more sensible default `LayerNormRule` would be nicer...

I think option 2 would be more sensible, but you should decide.

Reply:
I'm in favor of option 2: 
The documentation and README already strictly discourage the use of the `ZeroRule` and LRP without composite. And I rather like that LRP with the `ZeroRule` corresponds to `InputTimesGradient`. This would be broken by using a custom `ZeroRule` implementation for LayerNorm.